Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1163,25 +1332,82 @@ namespace Babylon
                 }
 
                 // Get the native results
-                std::vector<xr::Pose> nativeHitResults{};
+                std::vector<xr::HitResult> nativeHitResults{};
                 m_frame->GetHitTestResults(nativeHitResults, nativeRay);
 
                 // Translate those results into a napi array.
                 auto results = Napi::Array::New(info.Env(), nativeHitResults.size());
                 uint32_t i{0};
-                for (std::vector<xr::Pose>::iterator it = nativeHitResults.begin(); it != nativeHitResults.end(); ++it)
+                for (std::vector<xr::HitResult>::iterator it = nativeHitResults.begin(); it != nativeHitResults.end(); ++it)
                 {
                     Napi::Object currentResult = XRHitTestResult::New(info);
                     XRHitTestResult* xrResult = XRHitTestResult::Unwrap(currentResult);
-                    xrResult->SetPose(*it);
+                    xrResult->SetHitResult(*it);
+                    xrResult->SetXRFrame(this);
 
                     results[i++] = currentResult;
                 }
 
                 return results;
             }
+
+            Napi::Value CreateAnchor(const Napi::CallbackInfo& info)
+            {
+                XRRigidTransform* transform = XRRigidTransform::Unwrap(info[0].As<Napi::Object>());
+                return CreateNativeAnchor(info, transform->GetNativePose(), nullptr);
+            }
+
+            Napi::Value GetTrackedAnchors(const Napi::CallbackInfo& info)
+            {
+                // Create a JavaScript set to store all currently tracked anchors.
+                Napi::Object anchorSet = info.Env().Global().Get("Set").As<Napi::Function>().New({});
+
+                // Loop over the list of tracked anchors, and add them to the set.
+                for (Napi::Value napiValue : m_trackedAnchors)
+                {
+                    anchorSet.Get("add").As<Napi::Function>().Call(anchorSet, {napiValue});
```

Comment:
Can this just be:
```c++
anchorSet.Get("add").As<Napi::Function>()(anchorSet, {napiValue});
```

Reply:
I will give that a try, it was failing when I had tried just:
`anchorSet.Get("add").As<Napi::Function>()({napiValue});`

As it was not referencing back to anchorSet, and landed on the .call with the ref to the object, but looking at the operator what you suggested should work as well.

Reply:
Ahh nevermind the operator doesn't handle the receiver properly, will have to leave this using .Call.

    Value operator ()(const std::initializer_list<napi_value>& args) const;
vs:
    Value Call(napi_value recv, const std::initializer_list<napi_value>& args) const;